src/commands/add.ts

```diff
@@ -1,11 +1,44 @@
 import { SlashCommandBuilder } from "discord.js";
-import { CommandData } from "src/utils/cmdLoader.js";
+import { CommandData } from "../utils/cmdLoader.js";
+import { autoCompleteStreamers } from "../utils/cmdLoader.js";
+import { DiscordUser } from "../db/entity/DiscordUser.js";
+import { DiscordUserSubscription } from "../db/entity/DiscordUserSubscription.js";
+import { AppDataSource } from "../db/data-source.js";
+import { Streamer } from "../db/entity/Streamer.js";
+
+const command = new SlashCommandBuilder()
+  .setName("add")
+  .setDescription("Adds a user to your subscription list in your current channel.");
+// need to split this up for some reason
+command.addStringOption((option) =>
+  option
+    .setName("streamer")
+    .setDescription("The streamer to follow")
+    .setRequired(true)
+    .setAutocomplete(true)
+);
 
 const add: CommandData = {
-  command: new SlashCommandBuilder()
-    .setName("add")
-    .setDescription("Adds a user to your subscription list in your current channel."),
+  command: command,
+  autoComplete: autoCompleteStreamers,
   execute: async (interaction) => {
+    const discordUser = new DiscordUser(interaction.user.id);
+    await AppDataSource.getRepository(DiscordUser).save(discordUser);
+
+    const streamer = await AppDataSource.getRepository(Streamer).findOneOrFail({
+      where: {
+        name: interaction.options.getString("streamer", true), // TODO this is unsafe because name is not a primary key hence there could be multiple?
```

Comment:
ユニークインデックス追加すると問題なし。

Reply:
Unsure what you mean by this, the streamer table is in the README, the only unique key is the ID which is unobtainable from this context (since we're dealing with the interaction from autocomplete)

Reply:
マイグレーションで別のユニークインデックス追加する方が良いわけでした

Reply:
Fix in 59e3cb79b6dd9400b6c591813164a652b539df42